### DISCUSSION TITLE
Why can't use both classic bluetooth and ble?

### DISCUSSION THREAD

**Discussion by BrandonStudio** [PranjalKushwaha/ESP32-Bluetooth-FTP#3] [Q&A] [answered]
I'm attempting to use this repo on Arduino framework, but as [Espressif FAQ](https://espressif-docs.readthedocs-hosted.com/projects/arduino-esp32/en/latest/faq.html) says, I cannot easily modify `sdkconfig`, as a result it is hard to set bluetooth configs as the [README.md](https://github.com/PranjalKushwaha/ESP32-Bluetooth-FTP/blob/194127200689d6607d8f780ba865bc933f5f5fb4/README.md) says.

It comes to me that the bluetooth initialization failed, `esp_bt_controller_mem_release`, `esp_bt_controller_init`, and `esp_bt_controller_enable` returned `0x103`, i.e. `ESP_ERR_INVALID_STATE`.

I wonder is there a way to use this repo properly without rebuilding Arduino framework? (Arduino as component is out of question as it does not fit my requirement)

**Comment by BrandonStudio** [comment]
OK I know, I'll start a PR

**Reply by BrandonStudio** [reply]
See #4.

**Comment by PranjalKushwaha** [comment, marked as answer]
@BrandonStudio, thanks for adding C++ compatibility.

I have added Bluetooth dual-mode support. 
We cannot release BLE memory when the controller is in dual mode. Also, its init mode has to include BLE.
I have tested the code on ESP-IDF v5.1. I don't use the Arduino framework so please update me with the results. 

I apologise for the late reply, I was caught up in some uni assignments.
